[KB-276] Feedback (#177)
chhoumann/thesis-chemcam: report_thesis/src/sections/results/visual_analysis.tex

\subsection{Visual and Statistical Analysis of \ce{SiO_2} Distribution in Partitioned Data}\label{sec:visual_analysis}
This section provides a detailed visualization and statistical analysis of the \ce{SiO_2} concentration distribution across data partitions, following the customized k-fold data partitioning procedure described in Section~\ref{subsec:validation_testing_procedures}.

> We perform this analysis to validate the consistency of our data partitioning method and to provide a visual understanding of the data distribution it creates.

Reviewer: ```suggestion
This analysis is conducted to validate the consistency of our data partitioning method and to provide a visual understanding of the resulting data distribution.
```

The analysis focuses on \ce{SiO_2} as a representative example.

> We have conducted similar analyses for other oxides, but they are omitted here for brevity.

Reviewer: Should we mention that the figures for the other oxides can be found in the appendix? If we are still doing that.

Author: Not doing that for this section, IMO. It'll be 4 figures, spanning ~2 pages per oxide, which will be too much in the final report I think. Thoughts?

> We have conducted similar analyses for other oxides, but they are omitted here for brevity.

Reviewer: ```suggestion
Similar analyses have been conducted for other oxides, but they are omitted here for brevity.
```

> As discussed in Section~\ref{subsec:validation_testing_procedures}, it is crucial to determine an optimal value of $p$ for the data partitioning algorithm that minimizes the number of extreme values in the test set, while still maintaining its representativeness.

Reviewer: ```suggestion
As discussed in Section~\ref{subsec:validation_testing_procedures}, it is crucial to determine an optimal value of $p$ for the data partitioning algorithm that minimizes the number of extreme values in the test set while still maintaining its representativeness.
```

> As discussed in Section~\ref{subsec:validation_testing_procedures}, it is crucial to determine an optimal value of $p$ for the data partitioning algorithm that minimizes the number of extreme values in the test set while still maintaining its representativeness.

Reviewer 2: ```suggestion
As discussed in Section~\ref{subsec:validation_testing_procedures}, it is crucial to determine an optimal value of $p$ for the data partitioning algorithm.
This ensures that the number of extreme values in the test set is minimized, while still maintaining its representativeness.
```

Author: I understand that the sentence is a mouthful, but I think this would displace the meaning. What about this?

```suggestion
As mentioned in Section~\ref{subsec:validation_testing_procedures}, it is essential to identify an optimal value of $p$ for the data partitioning algorithm.
This value should minimize the number of extreme values in the test set while ensuring the test set remains representative.
```

> Our general approach is to select the lowest value of $p$ that still ensures the test set contains no extreme values, while maximizing its general representativeness.

Reviewer: ```suggestion
Our general approach is to select the lowest value of $p$ that ensures the test set contains no extreme values while maximizing its general representativeness.
```

> This helps in avoiding an overestimation of the model's performance due to extreme values while keeping the test set reflective of the majority of the data distribution.

Reviewer: ```suggestion
This approach helps avoid overestimating the model's performance due to extreme values while keeping the test set reflective of the majority of the data distribution.
```

> Our general approach is to select the lowest value of $p$ that ensures the test set contains no extreme values while maximizing its general representativeness.
> This approach helps avoid overestimating the model's performance due to extreme values while keeping the test set reflective of the majority of the data distribution.

Author: It's important to keep the term `general representativeness` because we define it in testing & validation.

> By general representativeness, we mean ensuring that the test set reflects the general distribution of the dataset without being skewed by extreme values.

It does get wordy, but this exact sentence structure is what we support. The words that are being removed aren't filler, but IMO necessary precision and nuance.

```suggestion
Our approach is to select the lowest $p$ that excludes extreme values from the test set while maximizing its general representativeness.
This approach helps avoid overestimating the model's performance due to extreme values while keeping the test set reflective of the majority of the data distribution.
```

We developed a web-based platform to evaluate the performance of the data partitioning algorithm for different values of $p$, as shown in Figure~\ref{fig:web_platform} in Appendix~\ref{subsec:web_platform}.

> Utilizing the platform, we conducted analyses for various values of $p$ and determined that an optimal value is $p=5\%$.

Reviewer: ```suggestion
Using the platform, we conducted analyses for various values of $p$ and determined that the optimal value is $p=5\%$.
```

> Furthermore, the 5\% threshold has demonstrated efficacy across all oxides within our dataset, and consequently, it has been uniformly applied throughout this study.

Reviewer: ```suggestion
Furthermore, the 5\% threshold has proven effective across all oxides within our dataset, and consequently, it has been uniformly applied throughout this study.
```

> This methodology is adaptable and can be employed to ascertain an optimal value for $p$ tailored to various targets, depending upon the specific dataset in use.

Reviewer: ```suggestion
This methodology is adaptable and can be employed to determine an optimal value for $p$ tailored to various targets, depending on the specific dataset in use.
```

Finally, we use a $k=5$ for our data partitioning algorithm.

> This results in five folds, with four folds used for cross-validation training and one fold designated as the test set.

Reviewer: Is there a reason to mention this again? This is the third time IIRC.

Author: This is an entirely different section, so it serves as a reminder for the user as context here. The sections are somewhat far apart, so I think it's OK to give a reminder.

Consequently, the full training set comprises the first four folds.

> Figures \ref{fig:histogram_grid_plot} and \ref{fig:histogram_kde_plot} illustrate the histograms and \gls{kde} curves for \ce{SiO_2} concentrations in each training fold, the test set, and their combined distributions.

Reviewer: ```suggestion
Figures~ref{fig:histogram_grid_plot} and \ref{fig:histogram_kde_plot} illustrate the histograms and \gls{kde} curves for \ce{SiO_2} concentrations in each training fold, the test set, and their combined distributions.
```

Author: ```suggestion
Figures~\ref{fig:histogram_grid_plot} and \ref{fig:histogram_kde_plot} illustrate the histograms and \gls{kde} curves for \ce{SiO_2} concentrations in each training fold, the test set, and their combined distributions.
```

The consistent histograms and \gls{kde} curves across different training folds indicate that the data distribution within each fold closely matches the overall distribution, confirming their consistency and representativeness.

> Figure \ref{fig:original_and_post_fold_plot} contrasts the \ce{SiO_2} concentration distribution before and after data partitioning.

Reviewer: ```suggestion
Figure~\ref{fig:original_and_post_fold_plot} contrasts the \ce{SiO_2} concentration distribution before and after data partitioning.
```

The left plot shows the original distribution, while the right plot displays the fold-assigned distribution, color-coded by fold.
This visualization highlights that the partitioning strategy maintains the overall data distribution while ensuring balanced representation across folds.

\begin{figure*}[h!]
    \centering
    \includegraphics[width=\textwidth]{images/histogram_grid_plot.png}

>     \caption{Histogram and \gls{kde} of \ce{SiO_2} Distribution in Each Fold. The y-axis represents the count of samples per bin, and the x-axis represents \ce{SiO_2} concentration. The notation in the legend indicates the amount of instances in the training/validation sets.}

Reviewer: ```suggestion
    \caption{Histogram and \gls{kde} of \ce{SiO_2} distribution in each fold. The y-axis represents the count of samples per bin, and the x-axis represents \ce{SiO_2} concentration. The notation in the legend indicates the amount of instances in the training/validation sets.}
```

    \label{fig:histogram_grid_plot}
\end{figure*}

\begin{figure*}[h!]
    \centering
    \includegraphics[width=\textwidth]{images/histogram_kde_plot.png}

>     \caption{Combined Histogram and \gls{kde} of \ce{SiO_2} Distribution in Each Fold. The y-axis represents the count of samples per bin, and the x-axis represents \ce{SiO_2} concentration. The notation in the legend indicates the amount of instances in the training/validation sets.}

Reviewer: ```suggestion
    \caption{Combined Histogram and \gls{kde} of \ce{SiO_2} distribution in each fold. The y-axis represents the count of samples per bin, and the x-axis represents \ce{SiO_2} concentration. The notation in the legend indicates the amount of instances in the training/validation sets.}
```

    \label{fig:histogram_kde_plot}
\end{figure*}

\begin{figure*}[h!]
    \centering
    \includegraphics[width=\textwidth]{images/original_and_post_fold.png}
    \caption{Distribution of \ce{SiO_2} concentrations before and after fold assignment. The left plot shows the original distribution of \ce{SiO_2}, while the right plot shows the distribution with folds assigned, color-coded to indicate the different folds.}
    \label{fig:original_and_post_fold_plot}
\end{figure*}

> To further validate our visual analysis, we can look at quantitative measures such as the means and standard deviations of \ce{SiO_2} concentrations across the folds and the overall dataset.

Reviewer: ```suggestion
To further validate our visual analysis, we can examine quantitative measures such as the means and standard deviations of \ce{SiO_2} concentrations across the folds and the overall dataset.
```

\begin{figure*}[htbp]
    \centering
    \includegraphics[width=\textwidth]{images/distribution_plot.png}
    \caption{Distribution of \ce{SiO_2} concentrations across cross-validation folds, training set, test set, and the entire dataset. The mean and standard deviation statistics for each partition are indicated figure.}
    \label{fig:siO2_distribution}
\end{figure*}

> From Figure~\ref{fig:siO2_distribution}, it is evident that the means and standard deviations of \ce{SiO_2} concentrations for each fold, as well as the combined training set, are consistent with those of the full dataset.

Reviewer: ```suggestion
Figure~\ref{fig:siO2_distribution} shows that the means and standard deviations of \ce{SiO_2} concentrations for each fold, as well as the combined training set, are consistent with those of the full dataset.
```

This quantitative consistency supports the visual evidence that each training fold is representative of the entire dataset.

> Furthermore, we can see that the standard deviation in the training sets is higher than the standard deviation in the test set, which is expected given the reassignment of extreme values to the training sets.

Reviewer: ```suggestion
Furthermore, we observe that the standard deviation in the training sets is higher than in the test set, which is expected given the reassignment of extreme values to the training sets.
```
